dashboard/app/assets/javascripts/header.js.erb

```diff
@@ -373,18 +373,30 @@ dashboard.header.showProjectHeader = function () {
   });
 
   $('.project_new').click(function (e) {
+    // TODO - this is currently broken
     e.preventDefault(); // Don't change the hash.
     // Save the project if one exists (i.e. isn't just the default one) then navigate.
     if (dashboard.project.getCurrentId()) {
       dashboard.project.save(function () {
-        location.href = location.pathname;
+        location.href = dashboard.project.appToProjectUrl();
       });
     } else {
-      location.href = location.pathname;
+      location.href = dashboard.project.appToProjectUrl();
     }
   });
 
   $(document).on('click', '.project_list', function () {
-    location.href = '/p'; <%# TODO: Can't call project_list_path because it's not in scope %>
+    location.href = '/projects'; <%# TODO: Can't call project_list_path because it's not in scope %>
   });
 };
+
+dashboard.header.updateTimestamp = function () {
```

Comment:
Moved this out of project. Trying to get to a place where project is mostly logic/state, and UI things happen more in header.